modules/apps/batch-planner/batch-planner-rest-impl/src/main/java/com/liferay/batch/planner/rest/internal/vulcan/batch/engine/FieldProvider.java

```diff
@@ -80,7 +82,25 @@ public List<Field> getFields(
 		Map<String, Field> fields = objectEntryOpenAPIResource.getFields(
 			uriInfo);
 
-		return new ArrayList<>(fields.values());
+		Map<String, Field> filteredFields = new HashMap<>();
+
+		for (Map.Entry<String, Field> entry : fields.entrySet()) {
+			Field field = entry.getValue();
+
+			if (Objects.equals(field.getType(), "object") ||
+				Objects.equals(field.getType(), "array")) {
+
+				String ref = field.getRef();
+
+				if ((ref != null) && ref.startsWith("#/components/schemas/"))
```

Comment:
Instead of the double validation, we could use the `startsWith` method of the `StringUtils`:

```
StringUtil.startsWith(field.getRef(), "#/components/schemas/")
```